libparsec/crates/platform_mountpoint/src/windows/mod.rs

```diff
@@ -0,0 +1,481 @@
+            Context::Dir(_) => unreachable!(),
+        })
+    }
+
+    fn write(
+        &self,
+        file_context: Self::FileContext,
+        buffer: &[u8],
+        offset: u64,
+        mode: WinWriteMode,
+    ) -> Result<usize, NTSTATUS> {
+        let fc = file_context.lock().expect("Mutex is poisoned");
+
+        let mode = WriteMode::from(mode);
+
+        if let Context::File(file) = fc.deref() {
+            Ok(self.fd_write(file.fd, buffer, offset, mode)?)
+        } else {
+            Err(STATUS_NOT_IMPLEMENTED)
+        }
+    }
+
+    fn flush(&self, file_context: Self::FileContext) -> Result<(), NTSTATUS> {
+        let fc = file_context.lock().expect("Mutex is poisoned");
+
+        if let Context::File(file) = fc.deref() {
+            self.fd_flush(file.fd);
+        }
+
+        Ok(())
+    }
+
+    fn set_basic_info(
+        &self,
+        _file_context: Self::FileContext,
+        _file_attributes: FileAttributes,
+        _creation_time: u64,
+        _last_access_time: u64,
+        _last_write_time: u64,
+        _change_time: u64,
+    ) -> Result<(), NTSTATUS> {
+        Err(STATUS_NOT_IMPLEMENTED)
+    }
+
+    fn set_file_size(
+        &self,
+        file_context: Self::FileContext,
+        new_size: u64,
+        set_allocation_size: bool,
+    ) -> Result<(), NTSTATUS> {
+        let fc = file_context.lock().expect("Mutex is poisoned");
+
+        if let Context::File(file) = fc.deref() {
+            self.fd_resize(file.fd, new_size, set_allocation_size)?;
+        }
+
+        Ok(())
+    }
+
+    fn rename(
+        &self,
+        _file_context: Self::FileContext,
+        file_name: &U16CStr,
+        new_file_name: &U16CStr,
+        replace_if_exists: bool,
+    ) -> Result<(), NTSTATUS> {
+        let path = PathBuf::from(file_name.to_os_string());
+        let new_path = PathBuf::from(new_file_name.to_os_string());
+        self.entry_rename(&path, &new_path, replace_if_exists)?;
+        Ok(())
+    }
+
+    fn get_security(
+        &self,
+        _file_context: Self::FileContext,
+    ) -> Result<PSecurityDescriptor, NTSTATUS> {
+        Ok(SD.as_ptr())
+    }
+
+    fn set_delete(
+        &self,
+        _file_context: Self::FileContext,
+        file_name: &U16CStr,
+        _delete_file: bool,
+    ) -> Result<(), NTSTATUS> {
+        let path = PathBuf::from(file_name.to_os_string());
+        let stat = self.entry_info(&path)?;
+
+        if path == Path::new("/") {
+            return Err(STATUS_RESOURCEMANAGER_READ_ONLY);
+        } else if !stat.children.is_empty() {
+            return Err(STATUS_DIRECTORY_NOT_EMPTY);
+        }
+
+        Ok(())
+    }
+}
+
+pub fn mount(mountpoint: &Path) -> anyhow::Result<FileSystem<MountpointManager>> {
+    let mountpoint =
+        U16CString::from_os_str(mountpoint.as_os_str()).expect("Unreachable, valid OsStr");
+    let mut volume_params = VolumeParams::default();
+
+    volume_params
+        .set_sector_size(512)
```

Comment:
From this comment https://github.com/winfsp/winfsp/issues/240#issuecomment-518629301

> The sector size is what is set in VolumeParams via the [SectorSize](https://github.com/billziss-gh/winfsp/blob/v1.5B2/src/dotnet/FileSystemHost.cs#L84) property when the file system is first created. (And of course the "sector size" is misnamed for non-disk file systems, but it determines the granularity at which non-buffered I/O is done.)

I think it would be good to add this comment (and the link) here, and also explain how we've choosen `512` as value (I guess we have just taken what is in winfsp example as it is a good default, which is totally a fine explanation :+1: )

Reply:
Done